### PR TITLE
Fix eating/drinking through helmets or masks + Limit to length of name and title on ID

### LINKED ISSUE #8278: Eating and drinking through helmets
### Environment
- Game Version: 22012917
- OS Version: Windows 10
- Playtest on January 29th 2022

### Detailed description of bug

People are able to eat and drink right through space-proof helmets which cover their faces entirely. See for example the Atmos hardsuit helmet. This likely ties in with Issue #5014.

### Steps to reproduce

1. Put on a hardsuit helmet.
2. Eat or drink!

### LINKED ISSUE #8327: ID job name lags server and game
### Environment
- Game Version: Windows 10
- OS Version: Staging 22020317

### Detailed description of bug

Basically, you can put an infinite number of characters in the Job name of your ID card as shown here,
Screenshot_186
Screenshot_185

This creates a decent amoung of lag when shown to other players, and even lagged the entire server for a few moments

### Steps to reproduce

1. Go to the ID Console
2. Change ID name to whatever, must be an insanely long name
3.  Profit, when showing it around its gonna lag the server and the game

### PR DESCRIPTION
**Pull request by ProudCipher**
### Purpose

Fixes #8278 
Fixes #8327 
### Notes:
This is accomplished by introducing a bool in ClothingV2.cs that is checked against in Equipment.cs, then the value is read by DrinkableContainer.cs and Edible.cs when attempting to consume the item. This bool has been set on the appropriate headwear. Masks, by default, will now prevent eating/drinking but an exception was made for the fake mustache item. 

Additionally, now when editing an ID through an ID console, name and title are limited to 32 characters.

### Changelog:
CL: [Fix] You can no longer eat or drink through helmets or masks that would usually prevent it.
CL: [Fix] When editing an ID through the ID console, name and job title are limited to 32 characters.

### CODE REVIEW

**Review by NoooneyDude — DISMISSED** [review]
Just so you're aware if you're not already, you just need to make the prefab change on the parent headwear / mask item. I think most apparel on these slots should block and the exception can be made for the things that don't block, like cigarettes.

Thanks for your contribution!

**Review by NoooneyDude — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]

**Comment by mmails** [comment]
You should put a note when you are trying to eat or consume something with a helmet/mask on

_Something is blocking your mouth, take it off before eating!_

**Comment by NoooneyDude** [comment]
There is a message posted, though your string does read nicer.

File: UnityProject/Assets/Scripts/UI/Objects/Command/IDConsole/GUI_IDConsole.cs [hunk 1]
```diff
@@ -147,14 +147,30 @@ private void ServerRefreshCardNames()
...
-			console.TargetCard.ServerSetRegisteredName(newName);
-			ServerRefreshCardNames();
+			if (newName.Length <= 32)
+			{
+				console.TargetCard.ServerSetRegisteredName(newName);
+				ServerRefreshCardNames();
+			}
+			else
+			{
+				Chat.AddExamineMsgToClient($"Name cannot exceed 32 characters!");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Objects/Command/IDConsole/GUI_IDConsole.cs:157]: This wont work, you'll need to do Chat.AddExamineMsg and pass the users

File: UnityProject/Assets/Scripts/UI/Objects/Command/IDConsole/GUI_IDConsole.cs [hunk 2]
```diff
@@ -147,14 +147,30 @@ private void ServerRefreshCardNames()
...
-			console.TargetCard.ServerSetJobTitle(newJobTitle);
-			ServerRefreshCardNames();
+			if (newJobTitle.Length <= 32)
+			{
+				console.TargetCard.ServerSetJobTitle(newJobTitle);
+				ServerRefreshCardNames();
+			}
+			else
+			{
+				Chat.AddExamineMsgToClient($"Job title cannot exceed 32 characters!");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Objects/Command/IDConsole/GUI_IDConsole.cs:171]: Same here

**Comment by corp-0** [comment]
@ProudCipher in the future try to separate unrelated stuff into their own PRs to simplify the review and merging process

**Pull request merged**